From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Todd Kjos <tkjos@google.com>
Date: Wed, 28 Aug 2019 15:52:02 -0700
Subject: ANDROID: init: GKI: add GKI_HACKS_TO_FIX

Add CONFIG_GKI_HACKS_TO_FIX as a mechanism to force
hidden configs to be selected for modules that will be built
separately. Also used to select drivers that need to be
modularized.

As these issues are resolved upstream, the configs should
be removed from GKI_HACKS_TO_FIX

Bug: 141266428
Change-Id: Ic8b2a17cd3a389ac5ef999c8c79b5b5dfee73c8a
Signed-off-by: Todd Kjos <tkjos@google.com>
---
 init/Kconfig     |  2 ++
 init/Kconfig.gki | 15 +++++++++++++++
 2 files changed, 17 insertions(+)
 create mode 100644 init/Kconfig.gki

diff --git a/init/Kconfig b/init/Kconfig
index b4daad2bac23..e3f4463ac31c 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -2258,3 +2258,5 @@ config ARCH_HAS_SYNC_CORE_BEFORE_USERMODE
 # <asm/syscall_wrapper.h>.
 config ARCH_HAS_SYSCALL_WRAPPER
 	def_bool n
+
+source "init/Kconfig.gki"
diff --git a/init/Kconfig.gki b/init/Kconfig.gki
new file mode 100644
index 000000000000..fafcd618037d
--- /dev/null
+++ b/init/Kconfig.gki
@@ -0,0 +1,15 @@
+# Atrocities needed for
+# a) building GKI modules in separate tree, or
+# b) building drivers that are not modularizable
+#
+# All of these should be reworked into an upstream solution
+# if possible.
+#
+config GKI_HACKS_TO_FIX
+	bool "GKI Dummy config options"
+	help
+	  Dummy config option used to enable core functionality used by
+	  modules that may not be selectable in this config.
+
+	  Unless you are building a GKI kernel to be used with modules
+	  built from a different config, say N here.
